File: evidence/generated-systems/angular/fintech-on-angular/src/app/models/Customer.js

// Define collection and schema for Customer
export interface Customer {
    firstName:
	type : string
    lastName:
	type : string
    dateOfBirth:
	type : Date
    email:
	type : Email
    phone:
	type : PhoneNumber
    address:
	type : Address
    taxId:
	type : TaxId
    riskScore:
	type : RiskScore
    Institution:
	type : Schema.Types.ObjectId
    Accounts:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Account' }]
    Wallets:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Wallet' }]
    Cards:
 	type : [{ type: Schema.Types.ObjectId, ref: 'PaymentCard' }]
    KycProfiles:
 	type : [{ type: Schema.Types.ObjectId, ref: 'KYCProfile' }]
    Consents:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Consent' }]
    Agreements:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Agreement' }]
    LoanApplications:
 	type : [{ type: Schema.Types.ObjectId, ref: 'LoanApplication' }]
    Loans:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Loan' }]
    Portfolios:
 	type : [{ type: Schema.Types.ObjectId, ref: 'InvestmentPortfolio' }]
    Disputes:
 	type : [{ type: Schema.Types.ObjectId, ref: 'Dispute' }]
    CustomerType:
 	type : String
#
    collection: 'customers'
}

File: evidence/generated-systems/django/fintech-on-django/fintechOnDjango/models/Customer.py
from django.db import models
from fintechOnDjango.models.CustomerType import CustomerType

#======================================================================
# 
# Encapsulates data for model Customer
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class Customer Declaration
#======================================================================
class Customer (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	firstName = models.CharField(max_length=200, null=True)
	lastName = models.CharField(max_length=200, null=True)
	dateOfBirth = models.DateField(null=True)
	email = Email
	phone = PhoneNumber
	address = Address
	taxId = TaxId
	riskScore = RiskScore
	institution = models.ForeignKey('FinancialInstitution', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	accounts = models.ManyToManyField('Account',  blank=True, related_name='+')
	wallets = models.ManyToManyField('Wallet',  blank=True, related_name='+')
	cards = models.ManyToManyField('PaymentCard',  blank=True, related_name='+')
	kycProfiles = models.ManyToManyField('KYCProfile',  blank=True, related_name='+')
	consents = models.ManyToManyField('Consent',  blank=True, related_name='+')
	agreements = models.ManyToManyField('Agreement',  blank=True, related_name='+')
	loanApplications = models.ManyToManyField('LoanApplication',  blank=True, related_name='+')
	loans = models.ManyToManyField('Loan',  blank=True, related_name='+')
	portfolios = models.ManyToManyField('InvestmentPortfolio',  blank=True, related_name='+')
	disputes = models.ManyToManyField('Dispute',  blank=True, related_name='+')
	customerType = models.CharField(max_length=64, null=True, choices=[(tag.name, tag.value) for tag in CustomerType])

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.firstName
		str = str + self.lastName
		str = str + self.dateOfBirth
		str = str + self.email
		str = str + self.phone
		str = str + self.address
		str = str + self.taxId
		str = str + self.riskScore
		str = str + self.customerType
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "Customer";
    
	def objectType(self):
		return "Customer";
